### PR TITLE
Update repo structure search 4

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Move Azure search to new sdk folder

### CODE REVIEW

**Comment by chidozieononiwu** [comment]
azp run net - search - ci

**Comment by chidozieononiwu** [comment]
azp run net - search - ci

**Comment by chidozieononiwu** [comment]
azp run net - search - ci

**Comment by chidozieononiwu** [comment]
@Azure/azure-sdk-eng

File: sdk/search/Build-SearchPackages.ps1 [hunk 1]
```diff
@@ -0,0 +1,9 @@
+$scriptDir = Split-Path $MyInvocation.MyCommand.Path -Parent
+$packagesDir = "$scriptDir\..\..\binaries\packages"
+
+if (!(Test-Path -Path $packagesDir)) {
+    mkdir $packagesDir
+}
+
+msbuild "$scriptDir\..\..\build.proj" /t:CreateNugetPackage /p:Scope="sdk\search"
+msbuild "$scriptDir\..\..\build.proj" /t:CreateNugetPackage /p:Scope="sdk\search"
```
Comment (weshaggard) [inline sdk/search/Build-SearchPackages.ps1]: These commands are duplicated. I think we will want to update the second command to build the data-plane libraries.

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Microsoft.Azure.Search.csproj [hunk 2]
```diff
@@ -1,24 +0,0 @@
...
-    <AssemblyName>Microsoft.Azure.Search</AssemblyName>
-    <Version>9.0.1</Version>
-    <PackageReleaseNotes>This is the newest major version of the Azure Search .NET SDK, based on version 2019-05-06 of the Azure Search REST API. New in this GA version is support for Cognitive Search, complex types for richer data modeling, the Autocomplete API, and additional parsing modes for Blob indexers. See this article for help on migrating to the latest version: http://aka.ms/search-sdk-upgrade. The 9.0.1 release includes a fix for the deadlock issue (https://github.com/Azure/azure-sdk-for-net/issues/6254) in the Search API.</PackageReleaseNotes>
-  </PropertyGroup>
-  <ItemGroup>
-    <Compile Include="..\AssemblyInfo.Common.cs" Link="Properties\AssemblyInfo.Common.cs" />
-  </ItemGroup>
-
-  <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.Search.Data" Version="[9.0.1, 10.0.0)" />
```
Comment (weshaggard) [inline src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Microsoft.Azure.Search.csproj]: @brjohnstmsft why did your old project have both PackageReference and ProjectReference to these libraries? It seems like that wouldn't have even worked.

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -1,11 +0,0 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License. See License.txt in the project root for
-// license information.
-
-using System.Reflection;
-
-[assembly: AssemblyVersion("9.0.0.0")]
```
Comment (weshaggard) [inline src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Properties/AssemblyInfo.cs]: @brjohnstmsft do you guys intentionally keep the AssemblyVersion fixed? By default when we start letting the SDK generate these it will generate them based on the Version in the project. However if you want to keep it fixed you can set another property to do that.

File: sdk/search/Microsoft.Azure.Search/tests/Tests/SynonymMapTests.cs [hunk 4]
```diff
@@ -265,6 +283,7 @@ public void ExistsReturnsTrueForExistingSynonymMap()
         }
 
         [Fact]
+		[LiveTest]
```
Comment (weshaggard) [inline sdk/search/Microsoft.Azure.Search/tests/Tests/SynonymMapTests.cs]: nit: indention

File: sdk/search/Microsoft.Azure.Search/tests/Tests/SkillsetsTests.cs [hunk 5]
```diff
@@ -255,6 +274,7 @@ public void ExistsReturnsFalseForNonExistingSkillset()
         }
 
         [Fact]
+		[LiveTest]
```
Comment (weshaggard) [inline sdk/search/Microsoft.Azure.Search/tests/Tests/SkillsetsTests.cs]: nit: indention

File: sdk/search/Microsoft.Azure.Search/tests/Tests/PostSuggestTests.cs [hunk 6]
```diff
@@ -5,110 +5,128 @@
...
 
         [Fact]
+		[LiveTest]
         public void CanSuggestWithCustomConverterViaSettings()
         {
             Run(TestCanSuggestWithCustomConverterViaSettings);
         }
 
         [Fact]
+		[LiveTest]
```
Comment (weshaggard) [inline sdk/search/Microsoft.Azure.Search/tests/Tests/PostSuggestTests.cs]: nit: indention

File: sdk/search/Microsoft.Azure.Search/tests/Tests/PostSearchTests.cs [hunk 7]
```diff
@@ -5,195 +5,227 @@
...
 
         [Fact]
+		[LiveTest]
         public void CanSearchWithCustomConverterViaSettings()
         {
             Run(TestCanSearchWithCustomConverterViaSettings);
         }
 
         [Fact]
+		[LiveTest]
```
Comment (weshaggard) [inline sdk/search/Microsoft.Azure.Search/tests/Tests/PostSearchTests.cs]: nit: indention

File: sdk/search/Microsoft.Azure.Search/tests/Tests/PostAutocompleteTests.cs [hunk 8]
```diff
@@ -4,113 +4,132 @@
...
 
         [Fact]
+		[LiveTest]
         public void CanAutocompleteWithFilter()
         {
             Run(TestAutocompleteWithFilter);
         }
 
         [Fact]
+		[LiveTest]
```
Comment (weshaggard) [inline sdk/search/Microsoft.Azure.Search/tests/Tests/PostAutocompleteTests.cs]: nit: indention

File: sdk/search/Microsoft.Azure.Search/tests/Tests/LookupTests.cs [hunk 9]
```diff
@@ -820,6 +838,7 @@ public void DynamicallyTypedPrimitiveCollectionsDoNotAllRoundtripCorrectly()
         }
 
         [Fact]
+		[LiveTest]
```
Comment (weshaggard) [inline sdk/search/Microsoft.Azure.Search/tests/Tests/LookupTests.cs]: nit: indention

File: sdk/search/Microsoft.Azure.Search/tests/Tests/IndexerTests.cs [hunk 10]
```diff
@@ -409,6 +431,7 @@ public void CanCreateBlobIndexerWithConfigurationParameters()
         }
 
         [Fact]
+		[LiveTest]
```
Comment (weshaggard) [inline sdk/search/Microsoft.Azure.Search/tests/Tests/IndexerTests.cs]: I will stop commenting on the indention in each file now. Please review and update all the test files you added the LiveTest attribute to and adjust the indention accordingly.

File: sdk/search/Microsoft.Azure.Management.Search/tests/Microsoft.Azure.Management.Search.Tests.csproj [hunk 11]
```diff
@@ -8,10 +8,19 @@
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <TargetFrameworks>netcoreapp2.0;net461</TargetFrameworks>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <ExcludeFromBuild>true</ExcludeFromBuild>
```
Comment (weshaggard) [inline sdk/search/Microsoft.Azure.Management.Search/tests/Microsoft.Azure.Management.Search.Tests.csproj]: @chidozieononiwu is there a reason you added these properties to exclude?

File: eng/mgmt/AzSdk.test.reference.props [hunk 12]
```diff
@@ -3,8 +3,8 @@
 		<TargetFrameworks>netcoreapp2.0</TargetFrameworks>
 	</PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.Test.HttpRecorder" Version="[1.13.3, 2.0.0)" />
-    <PackageReference Include="Microsoft.Rest.ClientRuntime.Azure.TestFramework" Version="[1.7.6, 2.0.0)" />
+    <PackageReference Include="Microsoft.Azure.Test.HttpRecorder" Version="[1.13.1, 2.0.0)" />
```
Comment (weshaggard) [inline eng/mgmt/AzSdk.test.reference.props]: We should update the versions in Packages.Data.props to match as opposed lowering this version.

Comment (chidozieononiwu) [reply eng/mgmt/AzSdk.test.reference.props]: I did that, but it did not turn out well as it broke many of the DataPlane projects using those packages.

File: sdk/search/Microsoft.Azure.Search/tests/Tests/IndexerTests.cs (hunk 10 above)
Comment (chidozieononiwu) [reply sdk/search/Microsoft.Azure.Search/tests/Tests/IndexerTests.cs]: I have those all resolved now.

File: sdk/search/Microsoft.Azure.Management.Search/tests/Microsoft.Azure.Management.Search.Tests.csproj (hunk 11 above)
Comment (chidozieononiwu) [reply sdk/search/Microsoft.Azure.Management.Search/tests/Microsoft.Azure.Management.Search.Tests.csproj]: There are failures in the test as it is not able to build for both net461 and netcore2.0 am looking into it. And I will file an issue if I cannot immediately resolve it.

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Microsoft.Azure.Search.csproj (hunk 2 above)
Comment (brjohnstmsft) [reply src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Microsoft.Azure.Search.csproj]: @weshaggard This goes back to a long conversation I had with @shahabhijeet last year. At that time, we were splitting what was just a single NuGet package into four packages (three component packages, and a meta-packages that includes those three). Abhijeet suggested using package references, but I found that I couldn't build in VS without project references. After some experimentation, we determined that if you have both, the project references take precedence at build time, which is what we wanted since we develop all these packages in tandem. They are build together, versioned together, and deployed together, always, much like the OData libraries (Microsoft.Spatial, Microsoft.OData.Edm, and Microsoft.OData.Core).

Is there a better way of handling this scenario?

Comment (weshaggard) [reply src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Microsoft.Azure.Search.csproj]: Given they always ship together I would just stick with ProjectReferences and remove the PackageReference, which is what we did in this PR. I just wanted to understand what you were trying to do before I approved that change.

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Properties/AssemblyInfo.cs (hunk 3 above)
Comment (brjohnstmsft) [reply src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Properties/AssemblyInfo.cs]: @weshaggard Our goal is to follow the SemVer rules as they apply to .NET, which means that the AssemblyVersion's major component should match the major version of the NuGet package, and the rest (minor, patch, and build numbers) should all be zero. Then to make the assemblies self-describing, we want the AssemblyFileVersion to reflect exactly the NuGet package version.

I'm not opinionated about how that policy is implemented, as long as it works consistently and predictably.

File: sdk/search/Microsoft.Azure.Search/tests/Tests/SynonymMapTests.cs (hunk 4 above)
Comment (brjohnstmsft) [reply sdk/search/Microsoft.Azure.Search/tests/Tests/SynonymMapTests.cs]: @weshaggard Why is this new attribute required?

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Properties/AssemblyInfo.cs (hunk 3 above)
Comment (weshaggard) [reply src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Properties/AssemblyInfo.cs]: I'm not aware of any SemVer rules that state that we should keep the assembly version parts to 0. That actually might cause problems if you add new APIs and people try to resolve your assembly in an application when 2 libraries use your libraries with the same assembly version but one depends on newer APIs then the other. I suggest we keep the assembly version, file version, and package version consistent unless there is a specific reason you need to make them differ.

File: sdk/search/Microsoft.Azure.Search/tests/Tests/SynonymMapTests.cs (hunk 4 above)
Comment (weshaggard) [reply sdk/search/Microsoft.Azure.Search/tests/Tests/SynonymMapTests.cs]: For our pipelines we filter tests based on category of "Live" and only run those nightly and not in CI. So my assumption is that these were live tests, correct @chidozieononiwu?

Comment (brjohnstmsft) [reply sdk/search/Microsoft.Azure.Search/tests/Tests/SynonymMapTests.cs]: All our tests use the Azure test framework and are capable of running in Playback mode. They should *all* run in CI, and should complete in a few seconds at most.

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Properties/AssemblyInfo.cs (hunk 3 above)
Comment (brjohnstmsft) [reply src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Properties/AssemblyInfo.cs]: We never make breaking changes without bumping the major version number, so the scenario you describe shouldn't happen. On the other hand, if we always force everyone to bind to a specific version of our assemblies, it makes dependency management very brittle since even a tiny point-release with non-breaking bug fixes would be treated as a completely different assembly (due to having a different strong name). So a customer using two libraries that each in turn take dependencies on different minor or patch versions of our assemblies would be out of luck. Am I not understanding that correctly?

Comment (weshaggard) [reply src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Properties/AssemblyInfo.cs]: It isn't about breaking changes. If you even add new APIs across minor or patch versions then you have this potential problem. I'm happy to chat and walk you through the those potential issues if you want to better understand. 

>a customer using two libraries that each in turn take dependencies on different minor or patch versions of our assemblies would be out of luck. 

They aren't out of luck they can correctly unify via binding redirects on .NET framework and just version in .NET Core. In fact if you you don't bump the version there is a good chance they may be out of luck because they might not have the fixes or new apis they expect or depend on.

File: sdk/search/Microsoft.Azure.Search/tests/Tests/SynonymMapTests.cs (hunk 4 above)
Comment (weshaggard) [reply sdk/search/Microsoft.Azure.Search/tests/Tests/SynonymMapTests.cs]: Ok I just chatted with @chidozieononiwu and I believe he was hitting those issues you called out in your email and thought it was because they were trying to run against the live service. He is going to undo the addition of these attributes and ensure all the tests pass for CI.

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Microsoft.Azure.Search.csproj (hunk 2 above)
Comment (brjohnstmsft) [reply src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Microsoft.Azure.Search.csproj]: As long as when we build, we get the version dependencies that we expect, I'm fine with dropping the package references.

File: sdk/search/Microsoft.Azure.Search/tests/Tests/SynonymMapTests.cs (hunk 4 above)
Comment (chidozieononiwu) [reply sdk/search/Microsoft.Azure.Search/tests/Tests/SynonymMapTests.cs]: I have gotten rid of the LiveTest Attibutes. I had to add the DisableTestRunParallel.cs import for dataplane tests to get it to build.

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Properties/AssemblyInfo.cs (hunk 3 above)
Comment (brjohnstmsft) [reply src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Properties/AssemblyInfo.cs]: I don't have a lot of experience with .NET Core, but binding redirects in .NET Framework have been the bane of my existence. If I understand correctly, that's all handled automatically in .NET Core, so now you can just do this:

```xml
<PropertyGroup>
  <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
  <GenerateBindingRedirectsOutputType>true</GenerateBindingRedirectsOutputType>
</PropertyGroup>
```

Is that right?

If this is the state of the art now, then I'm all in favor of not doing things differently. Go ahead and set our versions consistently with other Azure SDKs.

File: eng/.docsettings.yml [hunk 13]
```diff
@@ -154,6 +154,8 @@ known_presence_issues:
   - ['sdk/streamanalytics/Microsoft.Azure.Management.StreamAnalytics','#5499']
   - ['sdk/graphrbac/Microsoft.Azure.Graph.RBAC','#5499']
   
+  - ['sdk/search','#5499']
```
Comment (brjohnstmsft) [inline eng/.docsettings.yml]: What does this do?

File: eng/AzSdk.test.reference.targets [hunk 14]
```diff
@@ -12,4 +12,7 @@
     <!-- This is needed for discovering tests in test explorer -->
     <PackageReference Include="System.Runtime.InteropServices" />
   </ItemGroup>
+  <ItemGroup> 
+	   <Compile Include="$(RepoEngPath)\mgmt\DisableTestRunParallel.cs" Link="DisableTestRunParallel.cs" />
```
Comment (brjohnstmsft) [inline eng/AzSdk.test.reference.targets]: nit: indentation

File: eng/Packages.Data.props [hunk 15]
```diff
@@ -57,6 +57,7 @@
     <PackageReference Update="Microsoft.ServiceFabric.Services" Version="3.3.624" />
     <PackageReference Update="Microsoft.ServiceFabric" Version="6.4.624" />
     <PackageReference Update="Microsoft.SourceLink.GitHub" Version="1.0.0-beta2-18618-05" />
+    <PackageReference Update="Microsoft.Spatial" Version="7.5.3" />
```
Comment (brjohnstmsft) [inline eng/Packages.Data.props]: If I understand correctly, all dependencies will be centralized here from now on. Is the expectation that service teams are responsible for bumping the dependency version numbers in this file?

File: eng/mgmt/AzSdk.test.reference.props [hunk 16]
```diff
@@ -3,8 +3,8 @@
...
+    <PackageReference Include="Microsoft.Rest.ClientRuntime.Azure.TestFramework" Version="[1.7.5, 2.0.0)" />
```
Comment (brjohnstmsft) [inline eng/mgmt/AzSdk.test.reference.props]: Whose responsibility is it to update these version numbers, and how often are they expected to change?

File: sdk/search/Build-SearchPackages.ps1 [hunk 17]
```diff
@@ -0,0 +1,9 @@
+$scriptDir = Split-Path $MyInvocation.MyCommand.Path -Parent
+$packagesDir = "$scriptDir\..\..\binaries\packages"
+
+if (!(Test-Path -Path $packagesDir)) {
+    mkdir $packagesDir
+}
+
+msbuild "$scriptDir\..\..\build.proj" /t:CreateNugetPackage /p:Scope="sdk\search\Microsoft.Azure.Management.Search"
+msbuild "$scriptDir\..\..\build.proj" /t:CreateNugetPackage /p:Scope="sdk\search"
```
Comment (brjohnstmsft) [inline sdk/search/Build-SearchPackages.ps1]: So scope `sdk\search` is shorthand for the data plane? Is that a common pattern for other Azure services as well? It seems a bit counter-intuitive.

File: sdk/search/Microsoft.Azure.Management.Search/tests/Microsoft.Azure.Management.Search.Tests.csproj [hunk 18]
```diff
@@ -8,10 +8,19 @@
...
+    <ExcludeFromTest>true</ExcludeFromTest>
+  </PropertyGroup>
+
   <ItemGroup>
     <PackageReference Include="Microsoft.Azure.Management.ResourceManager" Version="1.5.0-preview" />
-    <ProjectReference Include="..\Management.Search\Microsoft.Azure.Management.Search.csproj" />
-    <ProjectReference Include="..\..\DataPlane\Microsoft.Azure.Search\Microsoft.Azure.Search.csproj" />
+    <ProjectReference Include="..\src\Microsoft.Azure.Management.Search.csproj" />
+    <ProjectReference Include="..\..\Microsoft.Azure.Search\src\Microsoft.Azure.Search.csproj" />
```
Comment (brjohnstmsft) [inline sdk/search/Microsoft.Azure.Management.Search/tests/Microsoft.Azure.Management.Search.Tests.csproj]: I'm not sure what dependency the management tests have on the data plane. I suspect this reference could be made more specific, maybe to Microsoft.Azure.Search.Common?

File: sdk/search/Microsoft.Azure.Search.Common/src/Microsoft.Azure.Search.Common.csproj [hunk 19]
```diff
@@ -1,15 +1,10 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
-  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
-  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSearch.DataPlane.props'))" />
   <PropertyGroup>
-    <PackageId>Microsoft.Azure.Search.Common</PackageId>
+    <PackageTags>Microsoft Azure Search;Search</PackageTags>
```
Comment (brjohnstmsft) [inline sdk/search/Microsoft.Azure.Search.Common/src/Microsoft.Azure.Search.Common.csproj]: Is the package ID derived from the project file name now?

File: sdk/search/Microsoft.Azure.Search.sln [hunk 20]
```diff
@@ -0,0 +1,49 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (brjohnstmsft) [inline sdk/search/Microsoft.Azure.Search.sln]: Our data plane tests rely on our management plane SDK, as well as some utilities defined in our management plane tests. I don't see those projects in this solution. Please make sure that loading and building the solution and running tests in Playback mode all continue to work.

File: sdk/search/Microsoft.Azure.Search/tests/Tests/FieldBuilderTests.cs [hunk 21]
```diff
@@ -345,8 +345,9 @@ private void OnlyTrueFor(Type modelType, Func<Field, bool> check, params string[
                 modelType,
                 fields =>
                 {
-                    foreach ((string fieldNameFromModel, Field field) in fields)
+                    foreach (string fieldNameFromModel in fields.Keys)
```
Comment (brjohnstmsft) [inline sdk/search/Microsoft.Azure.Search/tests/Tests/FieldBuilderTests.cs]: What did the usage of ValueTuple break (I'm ok with this change, just curious)?

File: sdk/search/Microsoft.Azure.Search/tests/Tests/Models/CustomAuthorConverter.cs [hunk 22]
```diff
@@ -64,7 +64,7 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
         {
             var author = (TAuthor)value;
 
-            string[] names = author.FullName.Split(' ', StringSplitOptions.RemoveEmptyEntries);
+            string[] names = author.FullName.Split(new char[] {' '}, StringSplitOptions.RemoveEmptyEntries);
```
Comment (brjohnstmsft) [inline sdk/search/Microsoft.Azure.Search/tests/Tests/Models/CustomAuthorConverter.cs]: Why was this change required?

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Properties/AssemblyInfo.cs [hunk 23]
```diff
@@ -1,11 +0,0 @@
...
-[assembly: AssemblyFileVersion("9.0.1.0")]
-
-[assembly: AssemblyTitle("Microsoft Azure Search Library")]
-[assembly: AssemblyDescription("Makes it easy to develop a .NET application that uses Azure Search.")]
```
Comment (brjohnstmsft) [inline src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Properties/AssemblyInfo.cs]: Where does `AssemblyDescription` come from now that these files are gone? That was useful when browsing NuGet packages IMO

**Review by bitsforthought — CHANGES_REQUESTED** [review]

File: eng/.docsettings.yml (hunk 13 above)
Comment (chidozieononiwu) [reply eng/.docsettings.yml]: It prevents the build from breaking in CI. There is a script that verifies the presence and content of ReadMes.md and also breaks the CI if there are issues with readMe. This lets us by pass it for now till readMes can be updated.

File: eng/Packages.Data.props (hunk 15 above)
Comment (weshaggard) [reply eng/Packages.Data.props]: The version will be centralized here to help align versions across the different sdk libraries. Service teams can update this but they should ping us to ensure that the version bump is ok and doesn't break things for others.

File: eng/mgmt/AzSdk.test.reference.props (hunk 16 above)
Comment (weshaggard) [reply eng/mgmt/AzSdk.test.reference.props]: Great question. Mostly they will be updated on-demand when someone needs a newer versions. However we are trying to come up with an update strategy for these to ensure we are staying up-to-date. We also have some static analysis tools that will flag any dependency that has security issues for example so we will update them when they get flagged.

File: sdk/search/Build-SearchPackages.ps1 (hunk 17 above)
Comment (weshaggard) [reply sdk/search/Build-SearchPackages.ps1]: For better or worse the management and data-plane toolsets are a little different, however if you are scoping to just search you can use create one solution that has all the projects and call `dotnet pack` on that solution to produce the packages for these libraries.

File: sdk/search/Microsoft.Azure.Search.Common/src/Microsoft.Azure.Search.Common.csproj (hunk 19 above)
Comment (weshaggard) [reply sdk/search/Microsoft.Azure.Search.Common/src/Microsoft.Azure.Search.Common.csproj]: Yes it is defaulted from the project name.

File: sdk/search/Microsoft.Azure.Management.Search/tests/Microsoft.Azure.Management.Search.Tests.csproj (hunk 18 above)
Comment (weshaggard) [reply sdk/search/Microsoft.Azure.Management.Search/tests/Microsoft.Azure.Management.Search.Tests.csproj]: That would be good to clean-up :)

File: sdk/search/Microsoft.Azure.Search/tests/Tests/FieldBuilderTests.cs (hunk 21 above)
Comment (weshaggard) [reply sdk/search/Microsoft.Azure.Search/tests/Tests/FieldBuilderTests.cs]: The reason is that your tests build and run on .NET 4.6.1 now as well and for that target it didn't support this new deconstruct language feature so we just expanded it out.

File: sdk/search/Microsoft.Azure.Search/tests/Tests/Models/CustomAuthorConverter.cs (hunk 22 above)
Comment (weshaggard) [reply sdk/search/Microsoft.Azure.Search/tests/Tests/Models/CustomAuthorConverter.cs]: This Split overload doesn't exist in .NET 4.6.1 (only on .NET Core) so we just needed to use the other overload.

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Properties/AssemblyInfo.cs (hunk 23 above)
Comment (weshaggard) [reply src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Properties/AssemblyInfo.cs]: It comes from the Description property in the project.

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Properties/AssemblyInfo.cs (hunk 3 above)
Comment (weshaggard) [reply src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Properties/AssemblyInfo.cs]: .NET Core doesn't use binding redirects it always just unifies to the highest loaded version. For .NET Framework the AutoGenerateBindingRedirects is a common way for applications to resolve the binding redirect problem and for newer tools it is on by default (https://github.com/dotnet/sdk/issues/1405). 

So my recommendation would be to use the same version for consistency and if we discover issues we can reconsider options.

File: sdk/search/Build-SearchPackages.ps1 (hunk 17 above)
Comment (chidozieononiwu) [reply sdk/search/Build-SearchPackages.ps1]: Since there are various data-plane packages I kept the solution file in the service directory for all data-plane. Then a separate one for the management plane which is in its package directory.

File: eng/Packages.Data.props (hunk 15 above)
Comment (brjohnstmsft) [reply eng/Packages.Data.props]: So if I'm not actively updating this and nobody else is using it, does anybody update it? If somebody else is using it and updates it, is my only chance to find out to watch all the PRs in the repo?

File: sdk/search/Build-SearchPackages.ps1 (hunk 17 above)
Comment (brjohnstmsft) [reply sdk/search/Build-SearchPackages.ps1]: What I'm trying to figure out is if we're doing something fundamentally different than other service teams, and if not, if this change reflects the same practice that others are using.

File: sdk/search/Microsoft.Azure.Search.Common/src/Microsoft.Azure.Search.Common.csproj (hunk 19 above)
Comment (brjohnstmsft) [reply sdk/search/Microsoft.Azure.Search.Common/src/Microsoft.Azure.Search.Common.csproj]: So let's say we decide to solve our GA-versus-preview problem by co-locating the code instead of maintaining a separate branch -- If we need to create a new project with a different project name, but the same package name, can we still use `PackageId` to accomplish that?

File: eng/.docsettings.yml (hunk 13 above)
Comment (brjohnstmsft) [reply eng/.docsettings.yml]: Is there a place I can go to learn more about these readme.md files and what they're supposed to do? If I understand correctly, this is a new thing (at least, Search never had one in this repo).

File: sdk/search/Build-SearchPackages.ps1 (hunk 17 above)
Comment (weshaggard) [reply sdk/search/Build-SearchPackages.ps1]: So far most service teams don't actively contribute to both their management and data-plane libraries at the same time so usually there isn't a solution that includes them both. However for you guys I have no problem with you have a solution that references them both and it will help with the scenario you are trying to accomplish with this script.

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Properties/AssemblyInfo.cs (hunk 3 above)
Comment (brjohnstmsft) [reply src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Properties/AssemblyInfo.cs]: As long as that's the standard practice, I'm fine with it. Thanks!

File: sdk/search/Microsoft.Azure.Search.Common/src/Microsoft.Azure.Search.Common.csproj (hunk 19 above)
Comment (weshaggard) [reply sdk/search/Microsoft.Azure.Search.Common/src/Microsoft.Azure.Search.Common.csproj]: the project name is the default for the package id but you can still explicitly set it if needed. However given that it is redundant here it is best to eliminate it to simply the project and prevent potential issues.

File: eng/Packages.Data.props (hunk 15 above)
Comment (weshaggard) [reply eng/Packages.Data.props]: Yes I think you have that correct. I suspect what most people will do is check their dependencies of the packages they are using before they release to ensure the package dependencies are understood and if there is some reason it was updated and you guys cannot take that update for some reason then you can always override the version in your project with "VersionOveride" metadata on your PackageReference. However the entire point of this mono repo is to try and enforce consistent dependencies so we should only vary in exceptional cases.

File: eng/.docsettings.yml (hunk 13 above)
Comment (chidozieononiwu) [reply eng/.docsettings.yml]: This might help [Analyze job](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki?pagePath=%2FThe%20Analyze%20Job%20Failed.%20What%20now%3F&pageId=5&wikiVersion=GBwikiMaster)

File: sdk/search/Build-SearchPackages.ps1 (hunk 17 above)
Comment (brjohnstmsft) [reply sdk/search/Build-SearchPackages.ps1]: We actually don't actively develop the data plane and management plane SDKs together (they are owned by separate teams); we just have these cross-project dependencies.

It's interesting... I would have assumed that every data plane's tests needed the management plane to provision the resources to be tested, so it surprises me that this isn't more common.

File: sdk/search/Microsoft.Azure.Search.Common/src/Microsoft.Azure.Search.Common.csproj (hunk 19 above)
Comment (brjohnstmsft) [reply sdk/search/Microsoft.Azure.Search.Common/src/Microsoft.Azure.Search.Common.csproj]: Agreed; Thanks for clarifying.

File: eng/Packages.Data.props (hunk 15 above)
Comment (brjohnstmsft) [reply eng/Packages.Data.props]: Makes sense. This is unlikely to be a problem in practice for us, at least for this particular dependency. I just want to make sure there is a process in place to detect and fix breakage quickly, since mono-repos don't have the benefit of isolation.

File: sdk/search/Build-SearchPackages.ps1 (hunk 17 above)
Comment (weshaggard) [reply sdk/search/Build-SearchPackages.ps1]: The majority of service teams are generally using static test resources as opposed to provisioning as part of the test steps, which is why most don't do that.

File: eng/.docsettings.yml (hunk 13 above)
Comment (brjohnstmsft) [reply eng/.docsettings.yml]: Thanks!

File: sdk/search/Microsoft.Azure.Search.sln (hunk 20 above)
Comment (chidozieononiwu) [reply sdk/search/Microsoft.Azure.Search.sln]: I have added management.test that to the solution file

**Review by bitsforthought — APPROVED** [review]
As soon as the build and tests are all working, this looks ready to go. Thanks!
